api/v1alpha4/condition_consts.go

```diff
@@ -218,3 +218,27 @@ const (
 	// WaitingForAvailableMachinesReason (Severity=Warning) reflects the fact that the required minimum number of machines for a machinedeployment are not available.
 	WaitingForAvailableMachinesReason = "WaitingForAvailableMachines"
 )
+
+// Conditions and condition Reasons for  MachineSets
+
+const (
+	// MachinesCreatedCondition documents that the machines controlled by the MachineSet are created.
+	// When this condition is false, it indicates that there was an error when cloning the infrastructure/bootstrap template or
+	// when generating the machine object.
+	MachinesCreatedCondition ConditionType = "MachinesCreated"
```

Comment:
Q: should this also represent machine are provisioning (not yet created)? Same should apply to the machine set scaling up (new machine being provisioned).
Q: is there a way to surface the machine set are scaling down?

Reply:
@fabriziopandini Thank you for the review. 

`MachinesCreatedCondition` only captures the creation of the Machine CR and associated dependencies (i.e. bootstrap and infra machine). This is the same assumption made in KCP. I've kept the two definitions identical. 
Scaling Down will be captured by the `ResizedCondition` as is [done with KCP](https://github.com/kubernetes-sigs/cluster-api/blob/f6fd5ed7dc0fa75788f76f17c64ae82976fcc70b/controlplane/kubeadm/controllers/status.go#L77-L79). I hope to work on that next. 
I've documented this in the issue tracker: https://github.com/kubernetes-sigs/cluster-api/issues/3486

Reply:
Ok, thanks for the explanation.
Then, should we add also Resized to be fully consistent with KCP?

Reply:
In this PR or follow up?

Reply:
In this PR, otherwise MachinesCreatedCondition alone doesn't surface the MS being scaling up or down

Reply:
@fabriziopandini I have added the `ResizedCondition` as well. Please have a look again.